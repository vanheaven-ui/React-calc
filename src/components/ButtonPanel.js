import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = buttonName => {
    clickHandler(buttonName);
  };

  const renderButton = name => (
    <Button
      ButtonName={name}
      color={['รท', 'x', '-', '+', '='].includes(name) ? 'default-btn-color' : 'grey-btn'}
      wide={name === '0'}
      clickHandler={name => handleClick(name)}
    />
  );
  return (
    <div className="btn-panel">
      <div className="btn-group">
        { renderButton('AC')}
        { renderButton('+/-')}
        { renderButton('%')}
        { renderButton('รท')}
      </div>
      <div className="btn-group">
        { renderButton('7')}
        { renderButton('8')}
        { renderButton('9')}
        { renderButton('x')}
      </div>
      <div className="btn-group">
        { renderButton('4')}
        { renderButton('5')}
        { renderButton('6')}
        { renderButton('-')}
      </div>
      <div className="btn-group">
        { renderButton('1')}
        { renderButton('2')}
        { renderButton('3')}
        { renderButton('+')}
      </div>
      <div className="btn-group">
        { renderButton('0')}
        { renderButton('.')}
        { renderButton('=')}
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
