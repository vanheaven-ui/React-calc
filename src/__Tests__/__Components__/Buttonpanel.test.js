import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import ButtonPanel from '../../components/ButtonPanel';
import '@testing-library/jest-dom';

const handleClick = () => '';

afterEach(cleanup);

test('is rendered without crashing', () => {
  const rootNode = document.createElement('div');
  ReactDOM.render(<ButtonPanel clickHandler={handleClick} />, rootNode);
});

test('doesnot change unexpectedly', () => {
  const buttonPanel = renderer.create(
    <ButtonPanel clickHandler={handleClick} />,
  );
  const tree = buttonPanel.toJSON();
  expect(tree).toMatchSnapshot();
});

test('is in the document', () => {
  render(<ButtonPanel clickHandler={handleClick} />);
  expect(screen.getByText('AC')).toBeInTheDocument();
  expect(screen.getByText('=')).toBeInTheDocument();
  expect(screen.getByText('+/-')).toBeInTheDocument();
  expect(screen.getByText('+')).toBeInTheDocument();
  expect(screen.getByText('%')).toBeInTheDocument();
  expect(screen.getByText('x')).toBeInTheDocument();
  expect(screen.getByText('.')).toBeInTheDocument();
  expect(screen.getByText('0')).toBeInTheDocument();
  expect(screen.getByText('1')).toBeInTheDocument();
  expect(screen.getByText('2')).toBeInTheDocument();
  expect(screen.getByText('3')).toBeInTheDocument();
  expect(screen.getByText('4')).toBeInTheDocument();
  expect(screen.getByText('5')).toBeInTheDocument();
  expect(screen.getByText('6')).toBeInTheDocument();
  expect(screen.getByText('7')).toBeInTheDocument();
  expect(screen.getByText('8')).toBeInTheDocument();
  expect(screen.getByText('9')).toBeInTheDocument();
  expect(screen.queryByText('()')).toBeNull();
  expect(screen.queryByText('10')).toBeNull();
});
