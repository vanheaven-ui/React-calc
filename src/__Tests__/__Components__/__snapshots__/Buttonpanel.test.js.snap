// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnot change unexpectedly 1`] = `
<div
  className="btn-panel"
>
  <div
    className="btn-group"
  >
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="default-btn-color default-btn"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="default-btn-color default-btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="default-btn-color default-btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="default-btn-color default-btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="grey-btn double-btn"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="grey-btn default-btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="default-btn-color default-btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
